Sources/Signal.swift

```diff
@@ -27,6 +27,12 @@ public final class Signal<Value, Error: Swift.Error> {
 	/// The state of the signal. `nil` if the signal has terminated.
 	private let state: Atomic<SignalState<Value, Error>?>
 
+	/// The status of the signal. It synchronizes with `sendLock`.
+	private var status: SignalStatus<Value, Error>
```

Comment:
Nice idea. Implemented in https://github.com/ReactiveCocoa/ReactiveSwift/pull/137/commits/ecab87173d8d1f436848f439e3058bdf895e368d. :tada:
Note that I have pulled in Read-Copy-Update from #139 to avoid introducing locks.